src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -648,6 +689,8 @@ void Simulator::initializeSensorData()
 
 void Simulator::pollForMAVLinkMessages(bool publish)
 {
+	_publish = publish;
```

Comment:
I'd change the interface to `void Simulator::pollForMAVLinkMessages()`, and add a separate setter for the flag `setPublish(bool publish)`. This way it's more explicit that you change the state of a class member.

Reply:
Thanks @bkueng.  I deviated slightly in naming since this creates a merge conflict with #11513. I went ahead and took care of the snake case in this method in my latest commit to this PR.  Also, I named the setter `publish_topics(const bool publish)` as it matches up with the pattern in `poll_topics()`.  If preferred, I can easily modify to your example names.  Let me know!

Reply:
>  I named the setter `publish_topics(const bool publish)`

With the current semantics I find that confusing as `publish_topics` does not publish anything.

>  Given the above, following this PR, it will be simple to collect all of the `orb_publish_auto()` calls and place them in a refactored `publish_topics()`

I don't think I'd do that as it forces you to store all data as class members. In terms of locality the current way is better, you directly see which data is published. Also it's not clear when to publish everything as data might come in at different rates. Or did I missunderstand your intent?

Reply:
No stress.  Pushed up.  I'll outline my thoughts in future work.  Thanks for the feedback!